core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java

```diff
@@ -88,7 +96,66 @@ public void sweep(String workflowId) {
             Monitors.error(CLASS_NAME, "sweep");
             LOGGER.error("Error running sweep for " + workflowId, e);
         }
+        if (workflow != null) {
+            unack(workflow);
+        } else {
+            queueDAO.setUnackTimeout(
+                    DECIDER_QUEUE, workflowId, properties.getWorkflowOffsetTimeout().toMillis());
+        }
+    }
+
+    @VisibleForTesting
+    void unack(WorkflowModel workflowModel) {
+        long postponeDurationSeconds = 0;
+        for (TaskModel taskModel : workflowModel.getTasks()) {
+            if (taskModel.getStatus() == Status.IN_PROGRESS) {
+                if (taskModel.getTaskType().equals(TaskType.TASK_TYPE_WAIT)
+                        || taskModel.getTaskType().equals(TaskType.TASK_TYPE_HUMAN)) {
+                    postponeDurationSeconds =
+                            (taskModel.getWaitTimeout() != 0)
+                                    ? taskModel.getWaitTimeout() + 1
+                                    : properties.getWorkflowOffsetTimeout().getSeconds();
+                } else {
+                    postponeDurationSeconds =
+                            (taskModel.getResponseTimeoutSeconds() != 0)
+                                    ? taskModel.getResponseTimeoutSeconds() + 1
+                                    : properties.getWorkflowOffsetTimeout().getSeconds();
+                }
+                break;
+            }
+            if (taskModel.getStatus() == Status.SCHEDULED) {
+                Optional<TaskDef> taskDefinition = taskModel.getTaskDefinition();
+                if (taskDefinition.isEmpty()) {
+                    taskDefinition =
```

Comment:
If task definition is empty within the task model, it would not exist within the workflow task either. You can skip the `isEmpty` check here